File: HomeStorage.Logic/Models/Product/ProductUpdateModel.cs
using Microsoft.AspNetCore.Http;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace HomeStorage.Logic.Models.Product
{
    public class ProductUpdateModel : ProductModel
    {
        [JsonIgnore]
        public IFormFile? NewImage { get; set; }
    }
}

File: frontend/src/models/Product/ProductUpdateModel.ts
import { IJsonWithFileModel } from "@/interfaces/IJsonWithFile";
import { ProductModel } from "./ProductModel";

export class ProductUpdateModel extends ProductModel implements IJsonWithFileModel {
    getFormData(): FormData {
        const formData = new FormData();
        if (this.newImage) {
            formData.append("file", this.newImage);
        }
        formData.append("json", JSON.stringify(this));

        return formData;
    }
    public newImage?: File;
}
